src/main.js

```diff
@@ -224,11 +233,42 @@ function fitCameraToSelection(offset = 1) {
     controls.maxDistance = distance * 2;
     controls.target = center;
 
+    // Check if objects is an empty array
+    if (Array.isArray(objects) && objects.length === 0) {
+        console.error('Error: Empty array provided to fitCameraToSelection');
+        return;
+    }
+
+    let newPosition;
+    if (orientation) {
+        switch (orientation) {
+            case "TOP":
+                newPosition = new THREE.Vector3(center.x, center.y, center.z + distance);
+                break;
+            case "SIDE":
+                newPosition = new THREE.Vector3(center.x, center.y + distance, center.z);
+                break;
+            case "FRONT":
+                newPosition = new THREE.Vector3(center.x + distance, center.y, center.z);
+                break;
+            default:
+                break;
+        }
+    } else {
+        newPosition = controls.target.clone().sub(direction);
+    }
+
+    timeline.clear();
+    timeline.fromTo(camera.position,
+        {x: camera.position.x, y: camera.position.y, z: camera.position.z},
+        {x: newPosition.x, y: newPosition.y, z: newPosition.z,
+            onUpdate: function() { controls.update(); }}
```

Comment:
You can use destructuring assignment here to make things a little less verbose.

```suggestion
        { ...camera.position },
        { ...newPosition,
            onUpdate: function() { controls.update(); }}
```